### PR TITLE
Backend changes for frontend stuff.

### PR DESCRIPTION
**Pull request by IsaacSahle**

### CODE REVIEW

File: golang/day_trader/src/buy.go [hunk 1]
```diff
@@ -75,9 +75,6 @@ func (buy *Buy) cancel(ctx context.Context, user *User) {
 
 func (buy *Buy) updateBuy(ctx context.Context) error {
 	_, err := db.Exec("update Buy set IntendedCashAmount=?, Price=?, ActualCashAmount=?, StockBoughtAmount = ? where Id=?", buy.IntendedCashAmount, buy.Price, buy.ActualCashAmount, buy.StockBoughtAmount, buy.Id)
-	if err != nil {
-		return err
-	}
```
Comment (Fattouche) [inline golang/day_trader/src/buy.go]: Not sure what I was doing here, thanks!

File: golang/day_trader/src/main.go [hunk 2]
```diff
@@ -94,19 +94,29 @@ var (
...
 
 func (s *server) Add(ctx context.Context, req *pb.Command) (*pb.BalanceResponse, error) {
 	user := getUser(req.UserId)
-	user.updateUserBalance(ctx, req.Amount, true)
-	return &pb.BalanceResponse{UserId: req.UserId}, nil
+	user, err := user.updateUserBalance(ctx, req.Amount, true)
+	if err == nil {
+		err = errors.New(fmt.Sprintf("Amount: %f Balance: %f", req.Amount, user.Balance))
+	}
+	return &pb.BalanceResponse{UserId: user.Id, Balance: user.Balance}, err
```
Comment (Fattouche) [inline golang/day_trader/src/main.go]: Maybe I am missing something but will this always return an error, one of which will contain non error information?

File: golang/day_trader/src/main.go [hunk 3]
```diff
@@ -237,14 +250,15 @@ func (s *server) SetSellAmount(ctx context.Context, req *pb.Command) (*pb.Respon
...
-		return &pb.Response{Message: trigger.toString()}, nil
+		createSellTrigger(ctx, req.UserId, req.Symbol, sell.Id, req.Amount)
+		return &pb.Response{UserId: user.Id, Message: "Successfully set sell amount"}, nil
 	}
 	err = trigger.updateCashAmount(ctx, req.Amount, user)
 	if err != nil {
 		logErrorEvent(ctx, err)
 	}
-	return &pb.Response{Message: trigger.toString()}, err
+	return &pb.Response{UserId: user.Id, Message: "Successfully set sell amount"}, err
```
Comment (Fattouche) [inline golang/day_trader/src/main.go]: I don't actually care about this but you don't think it is useful to tell the user what information is in their trigger?

File: golang/day_trader/src/main.go [hunk 4]
```diff
@@ -308,15 +322,15 @@ func (s *server) DumpLog(ctx context.Context, req *pb.Command) (*pb.Response, er
...
-	}
+	// conn, err := grpc.Dial(logUrl, grpc.WithInsecure())
+	// if err != nil {
+	// 	log.Printf("Failed to dial to %s with %v", logUrl, err)
+	// }
+	// client := pb.NewLoggerClient(conn)
+	// _, err = client.DisplaySummary(ctx, req)
+	// if err != nil {
+	// 	log.Println("DisplaySummary call to log server failed: ", err)
+	// }
```
Comment (Fattouche) [inline golang/day_trader/src/main.go]: Is this going to be implemented later on?

File: golang/day_trader/src/userStock.go [hunk 5]
```diff
@@ -12,9 +12,21 @@ func (userStock *UserStock) toString() string {
 	return string(bytes)
 }
 
+func getAllUserStock(userID string, stocks *map[string]int32) {
+	rows, _ := db.Query("SELECT StockSymbol, Amount from UserStock where UserId=?", userID)
+	defer rows.Close()
```
Comment (Fattouche) [inline golang/day_trader/src/userStock.go]: good use of defer here ;)

File: golang/workload_generator/workLoadGenerator.go [hunk 6]
```diff
@@ -46,45 +46,68 @@ var transactionNum = int32(1)
...
+		stockResp, err = client.CancelSetSell(ctx, command)
+		resp = stockResp.String()
 	case "DUMPLOG":
-		resp, err = client.DumpLog(ctx, command)
+		response, err = client.DumpLog(ctx, command)
+		resp = response.String()
 	case "DISPLAY_SUMMARY":
-		resp, err = client.DisplaySummary(ctx, command)
+		summaryResponse, err = client.DisplaySummary(ctx, command)
+		resp = summaryResponse.String()
```
Comment (Fattouche) [inline golang/workload_generator/workLoadGenerator.go]: thanks for changing this too!

File: golang/day_trader/src/main.go (hunk 2 above)
Comment (IsaacSahle) [reply golang/day_trader/src/main.go]: Oh yeah lol, I was testing something here. I've removed this.

File: golang/day_trader/src/main.go (hunk 3 above)
Comment (IsaacSahle) [reply golang/day_trader/src/main.go]: Yeah, that is a possibility. I didn't have a clear idea of how I wanted to implement that on the frontend, so I started simple. That does make sense though, I will make that change.

File: golang/day_trader/src/main.go (hunk 4 above)
Comment (IsaacSahle) [reply golang/day_trader/src/main.go]: Yeah, this stuff will be in my next change.

**Pull request merged**